cdi/src/main/java/org/infinispan/cdi/util/defaultbean/DefaultBeanExtension.java

```diff
@@ -340,11 +340,12 @@ void afterBeanDiscovery(@Observes AfterBeanDiscovery event, BeanManager manager)
             if (bean == null) {
                 bean = defaultProducerFields.get(e.getKey());
             }
-            if (duplicateDetectionMap.containsKey(e.getValue())) {
+            if (bean != null && duplicateDetectionMap.containsKey(e.getValue())) {
```

Comment:
@slaskawi yes, it happened when testing 2 uberjars together. I'm not 100% sure if this was the root cause (haven't had time to find a reproducer yet), but IMHO it's very likely to be so. I'm going to close this PR and I'll reopen when spot it again in supported use case (or found out, that the root cause was different)